tests/jballerina-unit-test/src/test/resources/test-src/query/query-exp-iterable-objects.bal

```diff
@@ -196,3 +197,32 @@ class IterableFromIterator {
 function getIterableObject(_Iterator iterator) returns IterableFromIterator {
     return new IterableFromIterator(iterator);
 }
+
+class MyIterable {
+    *object:Iterable;
+    public function iterator() returns object {
+                                           public function next() returns record {| int value; |}?;
+                                       } {
+        return new MyIterator();
+    }
+}
+
+int i = 0;
+
+public class MyIterator {
+    public function next() returns record {| int value; |}? {
+        i += 1;
+        if (i < 5) {
+            return {value: i};
+        }
+        return ();
+    }
+}
+
+public function testObjectIterator() {
```

Comment:
Shall we rename this to indicate that we're testing the iterator in a query expression?

Reply:
this is coming under the query expression tests anyway. please note the bal file name. WDYT?

Reply:
Yeah, I guess that's okay then.